src/commands/filter_parser.cc

```diff
@@ -410,20 +401,16 @@ absl::StatusOr<FilterParseResults> FilterParser::Parse() {
   }
   filter_identifiers_.clear();
   pos_ = 0;
-  VMSDK_ASSIGN_OR_RETURN(auto predicate, ParseExpression(0));
+  VMSDK_ASSIGN_OR_RETURN(auto parse_result, ParseExpression(0));
   if (!IsEnd()) {
     return UnexpectedChar(expression_, pos_);
   }
-  results.root_predicate = std::move(predicate);
+  results.root_predicate = std::move(parse_result.prev_predicate);
   results.filter_identifiers.swap(filter_identifiers_);
   // Log the built query syntax tree.
-  VMSDK_LOG(WARNING, nullptr) << "Parsed QuerySyntaxTree:";
-  bool valid = true;
-  PrintPredicate(results.root_predicate.get(), 0, true, valid);
-  // Temporary validation until we support all the new predicates
-  if (!valid) {
-    return absl::InvalidArgumentError("Unsupported query operation");
-  }
```

Comment:
If we are removing this, `"Unsupported query operation"` needs to be added in the `ParseUnquotedTextToken` function when detecting the FuzzyPredicate and the InfixPredicate